### PR TITLE
Create commentable concern and link person to comments model

### PR DESCRIPTION
**Pull request by abhishekbose87**
Hi @kgf, I am facing a issue wherein the specs are failing because I already see a text column `comments` and that is clashing with the `Comment` model I am introducing.
Is there a use for keeping the text field ?

Also can you confirm that The new comments model has the following db design. Hope I got that part of the requirements ready.

```
  create_table "comments", force: :cascade do |t|
    t.string   "title",            limit: 50, default: ""
    t.text     "description"
    t.integer  "commentable_id"
    t.string   "commentable_type"
    t.integer  "person_id"
    t.datetime "created_at",                               null: false
    t.datetime "updated_at",                               null: false
  end
```

_TODO_: Display comments made by person on the person view page.

### CODE REVIEW

**Comment by kgf** [comment]
Thanks for your help! 
Sorry, this was more complex than I had first imagined. What you can do is write a migration to move the existing comments to something like old _ comments. Ideally the migration would move the old into the new format, however, renaming the column and changing the form should clear your path.
On the schema in your comment, I don't think you need person _id column. That would be contained in the commendable _id.
Hope this helps, feel free to ask more.
Thanks,
Kevin

File: app/models/person.rb [hunk 1]
```diff
@@ -21,6 +22,7 @@ class Person < ActiveRecord::Base
   has_many :items, inverse_of: :owner, foreign_key: :owner_id
   has_many :inspections
   has_many :activities, as: :loggable
+  has_many :comments
```
Comment (kgf) [inline app/models/person.rb]: If you use the concern, won't that already have this ?

File: spec/models/comment_spec.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (kgf) [inline spec/models/comment_spec.rb:5]: Sounds like good advice

**Comment by kgf** [comment]
Perhaps you should get this working as polymorphic, then as a concern? I'll happily accept two PRs. The scope of the issue as I wrote it is too broad.

**Comment by abhishekbose87** [comment]
I guess you are saying that `Person` is commentable. Let me explain what I thought.
I was thinking that items, tasks will be commentable but those comments will be made by Person object. Does that make sense or did I totally go on the wrong track.

> What you can do is write a migration to move the existing comments to something like old _ comments. Ideally the migration would move the old into the new format

Could I remove the field altogether unless there is live data which we don't want to lose.

> Perhaps you should get this working as polymorphic, then as a concern? I'll happily accept two PRs. The scope of the issue as I wrote it is too broad.

One PR should be good. As you said, the scope was a little broad so I needed to get some feedback. But yeah we should be fine.

**Comment by kgf** [comment]
Sadly there is some live data, especially on the Person model that I would like to retain.
Person, Item et al are alike, they are `Commentable`

Does that help ?
Thanks
Kevin

**Comment by AliaTabassumAS** [comment]
.

**Comment by abhishekbose87** [comment]
@kgf Thanks for the info 👍 

I will update the PR by tomorrow. Let you know if I have more questions.

**Comment by abhishekbose87** [comment]
Hey @kgf, Can you have a look now and provide feedback.

Btw I really made heavy work at the end, by rebasing my branch with `master` (and it took so much time). After which I realised that `development` was the correct branch :facepalm:

File: app/views/people/_form.html.erb [hunk 3]
```diff
@@ -58,6 +58,6 @@
     <h4> Personnel Image </h4>
     <%= @person.portrait %>
     <%= f.file_field :portrait %>
-    <%= f.input :comments, :input_html => { :size => 40} %>
+    <%= f.input :old_comments, :input_html => { :size => 40} %>
```
Comment (kgf) [inline app/views/people/_form.html.erb:61]: Seems like you'll also need to add a table to display the new comments ?

File: spec/factories/comments.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+FactoryGirl.define do
+  factory :comment do
+
```
Comment (kgf) [inline spec/factories/comments.rb:3]: It would be good to have a valid factory

**Comment by kgf** [comment]
Thanks for all your help. Sorry to hear about the difficulties with merging. 

I made a couple of comments. I don't know if you have time to address them, or should I create new issues for them ?

**Pull request merged**